app-backend/tool/src/main/scala/eval/Interpreter.scala

```diff
@@ -92,10 +94,9 @@ object Interpreter extends LazyLogging {
     * without fetching any Rasters. Only interprets the structural validatity of
     * the AST, given the params.
     */
-  def interpretPure[M: Monoid](ast: MapAlgebraAST, params: EvalParams): Interpreted[M] = ast match {
-    case Source(id, label) if params.sources.isDefinedAt(id) => Valid(Monoid.empty)
-    case Source(id, _) => Invalid(NonEmptyList.of(MissingParameter(id)))
```

Comment:
Are missing params still checked for if this line is removed? In the case of a `Source` whose id isn't in the source mapping, the `case operation => ...` branch would call `foldMap` on an empty list, giving the zero-element of the `Monoid` (`Unit`, in all our usage sites). Wouldn't that mean the actual error of the params being missing is being swallowed?